Sources/Java2Swift/JavaToSwift.swift

```diff
@@ -58,55 +58,118 @@ struct JavaToSwift: ParsableCommand {
   )
   var input: String
 
+  /// Describes what kind of generation action is being performed by
+  /// Java2Swift.
+  enum GenerationMode {
+    /// Generate a configuration file given a Jar file.
+    case configuration(jarFile: String)
+
+    /// Generate Swift wrappers for Java classes based on the given
+    /// configuration.
+    case classWrappers(Configuration)
+  }
+
   mutating func run() throws {
-    var vmOptions: [String] = []
-    let classpath = classPathWithJarFile
-    if !classpath.isEmpty {
-      vmOptions.append("-cp")
-      vmOptions.append(contentsOf: classpath)
+    // Determine the mode in which we'll execute.
+    let generationMode: GenerationMode
+    if jarFile {
+      generationMode = .configuration(jarFile: input)
+    } else {
+      let config = try JavaTranslator.readConfiguration(from: URL(filePath: input))
+      generationMode = .classWrappers(config)
+    }
+
+    // Load all of the dependent configurations and associate them with Swift
+    // modules.
+    let dependentConfigs = try dependsOn.map { dependentConfig in
+      guard let equalLoc = dependentConfig.firstIndex(of: "=") else {
+        throw JavaToSwiftError.badConfigOption(dependentConfig)
+      }
+
+      let afterEqual = dependentConfig.index(after: equalLoc)
+      let swiftModuleName = String(dependentConfig[..<equalLoc])
+      let configFileName = String(dependentConfig[afterEqual...])
+
+      let config = try JavaTranslator.readConfiguration(from: URL(filePath: configFileName))
+
+      return (swiftModuleName, config)
     }
 
-    let jvm = try JavaVirtualMachine.shared(vmOptions: vmOptions)
-    try run(environment: jvm.environment())
+    // Form a class path from all of our input sources:
+    //   * Command-line option --classpath
+    var classPathPieces: [String] = classpath
+    switch generationMode {
+    case .configuration(jarFile: let jarFile):
+      //   * Jar file (in `-jar-file` mode)
```

Comment:
I didn't notice this before, `-jar` probably would spell nicer for the option btw.

Reply:
Sure! PR incoming